### PR TITLE
Set up pre-commit hooks

### PR DESCRIPTION
**Pull request by Calinou**
I tried to make `.clang-format` suit the current coding style as much as possible, but feel free to suggest tweaks anyway.

See https://pre-commit.com for more information.

### CODE REVIEW

File: src/thread_local.h [hunk 1]
```diff
@@ -19,22 +19,21 @@
...
 #ifndef thread_local
-# if __STDC_VERSION__ >= 201112 && !defined __STDC_NO_THREADS__
-#  define thread_local _Thread_local
-# elif defined _WIN32 && ( \
-       defined _MSC_VER || \
-       defined __ICL || \
-       defined __DMC__ || \
-       defined __BORLANDC__ )
-#  define thread_local __declspec(thread)
+#if __STDC_VERSION__ >= 201112 && !defined __STDC_NO_THREADS__
```
Comment (hpvb) [inline src/thread_local.h:22]: Not sure I like what happened here.

File: include/ppelib/ppelib-low-level.h [hunk 2]
```diff
@@ -20,8 +20,8 @@
 
 #include <ppelib/ppelib.h>
 
-ppelib_header_t* ppelib_get_header(ppelib_handle *handle);
```
Comment (hpvb) [inline include/ppelib/ppelib-low-level.h:23]: I personally prefer my pointers on the type as it's part of the type.

**Review by hpvb — CHANGES_REQUESTED** [review]
I am happy with the idea and the framework proposed but I don't think I like the actual .clang-format in here. Could you make a PR with the same but without the file changes and an empty clang format ruleset? I think changing the clang-format file while discussing on a PR is going to be very cumbersome for the both of us.

**Comment by Calinou** [comment]
@hpvb Done. I rebased this PR against the `main` branch as well.

**Review by hpvb — APPROVED** [review]

**Comment by hpvb** [comment]
Thanks!

**Comment by hpvb** [comment]
I found it easier to just integrate clang-format with eclipse and deal with the differences. I ended up merging your exact `.clang-format` into the refactor2 branch which should become main soon.

**Pull request merged**